### ISSUE TITLE
测试GenImage效果不佳

### ISSUE THREAD

**Issue by Vanleya** [chuangchuangtan/NPR-DeepfakeDetection#32]
求问大家我在测试GenImage时效果不好，会因为什么呢？以下是代码和结果

![Image](https://github.com/user-attachments/assets/bb6154f6-cf90-4a02-8c2a-47231a0d3999)

这个是test.py
`elif opt.test_mode == 'genimage':
    dataroot = opt.dir
    printSet(opt.test_mode)

    accs = []
    aps = []
    print(time.strftime("%Y_%m_%d_%H_%M_%S", time.localtime()))
    for method in os.listdir(dataroot):
        method_path = os.path.join(dataroot, method)
        if os.path.isdir(method_path):
            val_path = find_val_folder(method_path)
            if val_path:
                opt.dataroot = val_path
                opt.no_resize = False
                opt.no_crop = False
                acc, ap, _, _, _, _ = validate(model, opt)
                accs.append(acc)
                aps.append(ap)
                print("({} {:12}) acc: {:.2f}; ap: {:.2f}".format(method, method, acc * 100, ap * 100))
            else:
                print(f"Warning: No 'val' folder found for method {method}")
    print("({} {:10}) acc: {:.2f}; ap: {:.2f}".format('Mean', 'Mean', np.mean(accs) * 100, np.mean(aps) * 100))
    print('*' * 25)`

这个是datasets.py
`class GenImageDataset(Dataset):
    def __init__(self, opt, root):
        self.opt = opt
        self.root = root
        self.data = []
        self.label_dict = {'ai': 1, 'nature': 0}
        self._load_data()

    def _load_data(self):
        for method in os.listdir(self.root):
            method_path = os.path.join(self.root, method)
            if os.path.isdir(method_path):
                val_path = os.path.join(method_path, 'val')
                if os.path.exists(val_path):
                    for label_name in ['ai', 'nature']:
                        label_path = os.path.join(val_path, label_name)
                        if os.path.exists(label_path):
                            for img_name in os.listdir(label_path):
                                img_path = os.path.join(label_path, img_name)
                                self.data.append((img_path, self.label_dict[label_name]))

    def __len__(self):
        return len(self.data)

    def __getitem__(self, index):
        img_path, label = self.data[index]
        img = Image.open(img_path).convert('RGB')
        
        # 替换 resize 为 translate_duplicate
        if not self.opt.no_resize:
            img = translate_duplicate(img, self.opt.loadSize)
        
        if not self.opt.no_crop:
            img = TF.center_crop(img, (self.opt.cropSize, self.opt.cropSize))
        
        img = TF.to_tensor(img)
        img = TF.normalize(img, mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
        return img, label`
